crates/matrix-sdk/src/room/timeline/event_handler.rs

```diff
@@ -852,10 +855,17 @@ pub(crate) fn update_read_marker(
             // The read marker can't move backwards.
             if from < to {
                 let item = items.remove(from);
-                // Since the fully-read event's index was shifted to the left
-                // by one position by the remove call above, insert the fully-
-                // read marker at its previous position, rather than that + 1
-                items.insert(to, item);
+
+                // We don't want to re-insert the read marker if it is at the end of the
+                // timeline.
+                if to < items.len() {
+                    // Since the fully-read event's index was shifted to the left
+                    // by one position by the remove call above, insert the fully-
+                    // read marker at its previous position, rather than that + 1
+                    items.insert(to, item);
+                } else {
+                    *fully_read_event_in_timeline = false;
```

Comment:
But the event itself is in the timeline. Should we rename to `fully_read_marker_in_timeline` or something like that?

Reply:
This becomes tricky. This is to mark that the read marker should try to be updated when an event is received. So I guess something like `event_should_update_read_marker`?